### PR TITLE
chore: migrate flex in misc core/form

### PR DESCRIPTION
**Pull request by laurenashpole**
### Description

This PR migrates the Flex component from `@sanity/ui` v4 to v5 in the non-`inputs` directories in `packages/sanity/src/core/form`. It updates Flex across 21 files containing 34 JSX or styled-component instances. It's stacked on the https://github.com/sanity-io/sanity/pull/14364 to hopefully make QAing the form components easier.

### What to review

Make sure the prop updates look correct.

### Testing

I did manual QA in the Standard Inputs examples in Test Studio.

### Notes for release

N/A

---

> [!NOTE]
> **Low Risk**
> Mechanical import and prop renames across form UI with snapshot updates only; visual regression risk is limited to flex layout edge cases in breadcrumbs and asset lists.
> 
> **Overview**
> Migrates **`Flex`** in non-input `core/form` code from `@sanity/ui` to **`ui5`**, touching alerts, field headers, breadcrumbs, tag input, array error UI, and dataset/media-library asset dialogs.
> 
> Usage is updated to the v5 prop names: **`align` → `alignItems`**, **`justify` → `justifyContent`**, **`direction` → `flexDirection`**, and shorthand **`flex`** on rows is replaced with **`flexBasis` / `flexGrow`** where needed (e.g. asset list rows). **`FormBuilder`** snapshots reflect the new Flex DOM (extra min-size CSS variables).
> 
> No new behavior—layout and styling should match the previous `@sanity/ui` Flex after the prop mapping.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 61e1f163b9c14c94bcddb2e4ff5297face0b65eb. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/sanity/src/core/form/studio/__snapshots__/FormBuilder.test.tsx.snap [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html
 
-exports[`FormBuilder > should render a studio form (with tree editing dialog) 1`] = `"<div data-testid="field-title"><div data-ui="FormRow"><div><div style="--formFieldAnchorPath__kzylaa0: --input_title;"></div></div><div><div data-as="div" data-ui="Stack" data-level="0"><div data-as="div" data-ui="Flex"><div style="--min-height: 0; --min-width: 0; --flex-basis: 0%; --flex-grow: 1;" data-ui="fieldHeaderContentBox"><div data-as="div" data-ui="Stack"><div data-as="div" data-ui="Flex"><div data-as="div" data-ui="Flex"><label data-ui="Text" for="title"><span>Title</span></label></div></div></div></div></div><div><div><div data-testid="change-bar-wrapper"><div data-testid="change-bar__field-wrapper"><span data-as="span" data-ui="TextInput" data-scheme="light" data-tone="default"><span><input data-as="input" data-scheme="light" data-tone="default" id="title" autocomplete="off" style="anchor-name: --input_title;" type="text" value=""><span data-border="" data-scheme="light" data-tone="default"></span></span></span></div></div></div></div></div></div></div></div>"`;
```
Comment (laurenashpole) [inline packages/sanity/src/core/form/studio/__snapshots__/FormBuilder.test.tsx.snap:3]: Updated this snapshot because the className and style differences are expected.

File: packages/sanity/src/core/form/components/formField/FormFieldSetLegend.tsx [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import {ToggleArrowRightIcon} from '@sanity/icons/ToggleArrowRight'
-import {Flex, rem, Text, type Theme} from '@sanity/ui'
+import {rem, Text, type Theme} from '@sanity/ui'
 import {memo, type ReactNode} from 'react'
 import {css, styled} from 'styled-components'
-import {Box} from 'ui5'
+import {Flex, Box} from 'ui5'
```
[the remark this answers is not in the document]
Comment (laurenashpole) [reply packages/sanity/src/core/form/components/formField/FormFieldSetLegend.tsx:5]: `forwardedAs` comes from the styled-components API not the Flex component (in either version v4 or v5) so this migration shouldn't impact it at all. I manually tested a collapsible fieldset legend to double check that it still renders as a button.

**Review by stipsan — DISMISSED** [review]
Sweet!

**Comment by laurenashpole** [comment]
@stipsan I had to resolve merge conflicts in this PR and lost your approval. Would you mind re-approving?

**Review by stipsan — APPROVED** [review]
